### PR TITLE
chore: deploy workspace proxies on fly.io

### PR DESCRIPTION
**Pull request by matifali**
This pull request adds a new job called `deploy-wsproxies` thst is responsible for deploying the workspace proxies to the fly.io platform. The deployment is triggered only when the pull request is made from the `main` branch.

The `deploy-wsproxies` job updates our workspaces proxies in 3 fly.io regions

1. Paris
2. Sydney
3. Sao Paulo

This pull request ensures that the proxies are up-to-date with `main` btanch.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -767,6 +767,27 @@ jobs:
           set -euxo pipefail
           kubectl --namespace coder rollout restart deployment/coder
           kubectl --namespace coder rollout status deployment/coder
+  
+  deploy-wsproxies:
+    runs-on: ubuntu-latest
+    needs: build
+    if: !github.event.pull_request.head.repo.fork
```
Comment (deansheather) [inline .github/workflows/ci.yaml:774]: Remember to add main check

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -767,6 +767,27 @@ jobs:
...
+          fetch-depth: 0
+
+      - name: Setup flyctl
+        uses: superfly/flyctl-actions/setup-flyctl@master
+
+      - name: Deploy workspace proxies
+        run: |
+          flyctl deploy --image ghcr.io/coder/coder-preview:main --app paris-coder -y
+          flyctl deploy --image ghcr.io/coder/coder-preview:main --app sydney-coder -y
+          flyctl deploy --image ghcr.io/coder/coder-preview:main --app sao-paulo-coder -y
```
Comment (deansheather) [inline .github/workflows/ci.yaml:788]: Does this automatically re-pull? a la `imagePullPolicy: Always`?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -767,6 +767,27 @@ jobs:
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup flyctl
+        uses: superfly/flyctl-actions/setup-flyctl@master
+
+      - name: Deploy workspace proxies
```
Comment (deansheather) [inline .github/workflows/ci.yaml:784]: Can we store manifests for these in the repo? In case we need to change proxy env vars perhaps

Comment (matifali) [reply .github/workflows/ci.yaml:784]: yes we can store. That will require encrypting secrets or domain host names.e.g (`PROXY_TOKEN`)

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (matifali) [reply .github/workflows/ci.yaml:788]: Good catch. No it does not unless the tag changes
I am changing it to `--image ghcr.io/coder/coder-preview:$tag`

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (matifali) [reply .github/workflows/ci.yaml:774]: We can add, but I believe this is a duplication as the `build` job already has the `main` check.

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (deansheather) [reply .github/workflows/ci.yaml:784]: Can those secrets just get stored in a "secrets" tab in fly.io? IDK how fly works so maybe that doesn't exist

Comment (matifali) [reply .github/workflows/ci.yaml:784]: Yes we can do that too.

**Review by deansheather — APPROVED** [review]
meant to approve that, please see comments above

**Pull request merged**